src/formatDate.js

```diff
@@ -50,7 +50,49 @@
  */
 
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const oldSeparator = fromFormat[3];
+  const newSeparator = toFormat[3];
+  const oldDate = date.split(oldSeparator);
+  const newFormatOfDate = [];
+  const indexOfDay = fromFormat.indexOf('DD');
+  const indexOfMonth = fromFormat.indexOf('MM');
+  let indexOfYear = fromFormat.indexOf('YY');
+
+  if (indexOfYear === -1) {
+    indexOfYear = fromFormat.indexOf('YYYY');
+  }
+
+  let year = oldDate[indexOfYear].slice(-2);
+  const month = oldDate[indexOfMonth];
+  const day = oldDate[indexOfDay];
+
+  for (let i = 0; i < toFormat.length - 1; i++) {
+    switch (toFormat[i]) {
+      case 'YY':
+        newFormatOfDate[i] = year;
+        break;
+
+      case 'YYYY':
+        if (+year < 30) {
+          year = '20' + year;
+        } else {
+          year = '19' + year;
+        }
+
+        newFormatOfDate[i] = year;
+        break;
+
+      case 'MM':
+        newFormatOfDate[i] = month;
+        break;
+
+      case 'DD':
+        newFormatOfDate[i] = day;
+        break;
+    }
```

Comment:
All `case` blocks must have a `break` statement. Also, the `switch/case` statement always must have a `default` block. In the `default` block, we usually raise exceptions if something went wrong. formatDate function can return a string "Get wrong date format" in the `default` block